src/main/java/org/mskcc/cbio/portal/scripts/ImportTabDelimData.java

```diff
@@ -332,6 +393,60 @@ public void importData(int numLines) throws IOException, DaoException {
         }
     }
 
+    private void expandRemainingGeneticEntityTabDelimitedRowsWithBlankValue() {
+        if (updateMode) {
+            // Expand remaining genetic entity id rows that were not mentioned in the file
+            new HashSet<>(geneticAlterationMap.keySet()).forEach(geneticEntityId -> {
+                try {
+                    String[] values = new String[orderedImportedSampleList.size()];
+                    Arrays.fill(values, "");
+                    saveValues(geneticEntityId, values);
+                } catch (DaoException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        }
+    }
+
+    private void ensureSampleGeneticProfile(Sample sample) throws DaoException {
+        if (!DaoSampleProfile.sampleExistsInGeneticProfile(sample.getInternalId(), geneticProfileId)) {
+            Integer genePanelID = (genePanel == null) ? null : GeneticProfileUtil.getGenePanelId(genePanel);
+            if (updateMode) {
+                DaoSampleProfile.deleteRecords(List.of(sample.getInternalId()), List.of(geneticProfileId));
+            }
+            DaoSampleProfile.addSampleProfile(sample.getInternalId(), geneticProfileId, genePanelID);
+        }
+    }
+
+    private void saveOrderedSampleList() throws DaoException {
+        if (updateMode) {
+            ArrayList <Integer> savedOrderedSampleList = DaoGeneticProfileSamples.getOrderedSampleList(geneticProfileId);
+            int initialOrderSampleListSize = savedOrderedSampleList.size();
+            checkSamplesInDataEqualTo(initialOrderSampleListSize);
+            // add all new sample ids at the end
+            ArrayList<Integer> extendedSampleList = new ArrayList<>(savedOrderedSampleList);
+            List<Integer> newSampleIds = orderedSampleList.stream().filter(sampleId -> !savedOrderedSampleList.contains(sampleId)).toList();
```

Comment:
why do we filter? Should we not remove the sample data and just load as new?

Reply:
And how do we reconcile this with the result of `filterOutNormalValues`?

Reply:
See the new approach implemented in `GeneticAlterationImporter` classes.